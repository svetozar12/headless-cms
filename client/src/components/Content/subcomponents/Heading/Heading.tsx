import { Breadcrumbs, Link, Typography } from "@mui/material";
import { useRouter } from "next/router";
import React, { FC } from "react";
import { FaBoxes } from "react-icons/fa";
import { CONTENT_LIST } from "../../../../constants/routes";
import Heading from "../../../Heading";
import PageHeader from "../../../PageHeader";
import Spinner from "../../../Spinner";

interface IModelHeading {
  title: string;
  isLoading: boolean;
}

const ModelHeading: FC<IModelHeading> = ({ title = "", isLoading = false }) => {
  const router = useRouter();
  return (
    <PageHeader extraProps={{ className: "justify-between relative" }}>
      <Spinner isLoading={isLoading} />
      <div className="flex">
        <FaBoxes className="mr-4 h-8 w-8" />
        <Breadcrumbs className="text-white" aria-label="breadcrumb">
          <Link
            underline="hover"
            className="cursor-pointer text-cms_blue-100 text-2xl font-bold"
            onClick={() => router.push(CONTENT_LIST)}
          >
            Content
          </Link>
          <p className="text-2xl font-bold">{title}</p>
        </Breadcrumbs>
      </div>
    </PageHeader>
  );
};

export default ModelHeading;
