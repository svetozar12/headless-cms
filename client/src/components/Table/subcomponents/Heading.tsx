import React, { FC } from "react";
import { IColumn } from "../Table";

const Heading: FC<{ columns: IColumn[] }> = (props) => {
  const { columns } = props;
  return (
    <tr className={`bg-off flex py-2 text-gray-400`}>
      {columns.map(({ title }) => (
        <th key={title} className="flex-1 text-center font-semibold">
          {title}
        </th>
      ))}
    </tr>
  );
};

export default Heading;
