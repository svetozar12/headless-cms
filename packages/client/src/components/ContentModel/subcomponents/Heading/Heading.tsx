import React, { FC } from "react";
import { FaBoxes } from "react-icons/fa";
import { Heading, PageHeader } from "@headless-cms/ui";

interface IModelHeading {
  title: string;
  ActionButtons: JSX.Element;
}

const ModelHeading: FC<IModelHeading> = (props) => {
  const { title, ActionButtons } = props;

  return (
    <PageHeader extraProps={{ className: "justify-between" }}>
      <div className="flex">
        <FaBoxes className="mr-4 h-8 w-8" />
        <Heading type="h1" text={title} />
      </div>
      {ActionButtons}
    </PageHeader>
  );
};

export default ModelHeading;
