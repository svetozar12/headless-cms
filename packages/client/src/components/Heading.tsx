interface IHeadingProps {
  text: string;
  type: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
}

const Heading = (props: IHeadingProps): JSX.Element => {
  const { text, type } = props;

  const getHeading = (): JSX.Element => {
    const headingClassname = "text-white font-bold text-2xl";
    switch (type) {
      case "h1":
        return <h1 className={headingClassname}>{text}</h1>;
      case "h2":
        return <h2 className={headingClassname}>{text}</h2>;
      case "h3":
        return <h3 className={headingClassname}>{text}</h3>;
      case "h4":
        return <h4 className={headingClassname}>{text}</h4>;
      case "h5":
        return <h5 className={headingClassname}>{text}</h5>;
      case "h6":
        return <h6 className={headingClassname}>{text}</h6>;
    }
  };
  return <div className="flex items-center justify-center">{getHeading()}</div>;
};

export default Heading;
