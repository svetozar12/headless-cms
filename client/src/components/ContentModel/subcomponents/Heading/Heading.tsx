import Breadcrumbs from "@mui/material/Breadcrumbs";
import Link from "@mui/material/Link";
import { useRouter } from "next/router";
import React, { FC } from "react";
import { FaBoxes } from "react-icons/fa";
import { MODEL_LIST } from "../../../../constants/routes";
import Heading from "../../../Heading";
import PageHeader from "../../../PageHeader";

interface IModelHeading {
  title: string;
  ActionButtons: JSX.Element;
}

const ModelHeading: FC<IModelHeading> = ({ title, ActionButtons }) => {
  const router = useRouter();
  return (
    <PageHeader extraProps={{ className: "justify-between" }}>
      <div className="flex">
        <FaBoxes className="mr-4 h-8 w-8" />
        <Breadcrumbs className="text-white" aria-label="breadcrumb">
          <Link
            underline="hover"
            className="cursor-pointer text-cms_blue-100 text-2xl font-bold"
            onClick={() => router.push(MODEL_LIST)}
          >
            Model
          </Link>
          <p className="text-2xl font-bold">{title}</p>
        </Breadcrumbs>
      </div>
      {ActionButtons}
    </PageHeader>
  );
};

export default ModelHeading;
