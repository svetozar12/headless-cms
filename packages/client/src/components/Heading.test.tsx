import type { ReactWrapper } from "enzyme";
import { mount } from "enzyme";
import { beforeEach, describe, expect, it } from "vitest";
import Heading from "./Heading";

const headingTypes = ["h1", "h2", "h3", "h4", "h5", "h6"];

describe("Button component", () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(<Heading type="h2" text="test" />);
  });
  it("should render button", () => {
    expect(wrapper.find(Heading).length).toBe(1);
    expect(wrapper.find("h2").length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
  headingTypes.forEach((type) => {
    it(`should render ${type}`, () => {
      wrapper.setProps({ type });
      expect(wrapper.find(type).length).toBe(1);
      expect(wrapper.find(type).text()).toBe("test");
    });
  });
});
