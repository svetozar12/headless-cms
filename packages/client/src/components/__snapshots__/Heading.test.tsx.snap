// Vitest Snapshot v1

exports[`Button component > should render button 1`] = `ReactWrapper {}`;
